File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F16.P2.cpp
#include <cstdint>
#include <cstdio>
#include <map>
#include <string>

std::int32_t map_total() {
// BEGIN PROBE F16.P2
std::map<std::string, std::int32_t> mp{{"a", 1}};
std::int32_t total = 0;
for (const auto& e : mp) total += e.second;
std::int32_t miss = mp.contains("b") ? mp.at("b") : 0;
return total + miss;
// END PROBE F16.P2
}

int main() {
    std::printf("%d\n", map_total());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F16.P2.ts
function mapTotal(): number {
// BEGIN PROBE F16.P2
const mp = new Map([["a", 1]]);
let total = 0;
for (const [, v] of mp) total += v;
const miss = mp.get("b") ?? 0;
return total + miss;
// END PROBE F16.P2
}

console.log(mapTotal());
